crates/matrix-sdk-sled/src/state_store.rs

```diff
@@ -426,6 +429,12 @@ impl SledStore {
                                 self.serialize_event(&event)
                                     .map_err(ConflictableTransactionError::Abort)?,
                             )?;
+                            // clear out any remaining stripped data
+                            stripped_members.remove(self.encode_key(STRIPPED_ROOM_MEMBER, &key))?;
```

Comment:
Are we starting to discard stripped state because of this sentence:

> Clients should only use stripped state events so long as they don’t have access to the proper state of the room. Once the state of the room is available, all stripped state should be discarded. In cases where the client has an archived state of the room (such as after being kicked) and the client is receiving stripped state for the room, such as from an invite or knock, then the stripped state should take precedence until fresh state can be acquired from a join.

If so, what about `striped_rooms` and `stripped_state`.

Reply:
This was removed as we have, on a higher level, a mixing of looking up that data. To ensure we don't accidentally return bad stale state, I added some code here to clear the opposing index.